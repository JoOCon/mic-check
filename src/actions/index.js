export const addUser = (user) => ({
  type: 'ADD_USER',
  user
});

export const getLocalStorageUsers = (users) => ({
  type: 'GET_LOCAL_STORAGE_USERS',
  users
});

export const setActiveUser = (user) => ({
  type: 'SET_ACTIVE_USER',
  user
});

export const logoutActiveUser = () => ({
  type: 'LOGOUT_ACTIVE_USER'
});

export const getLocalStorageRentals = (rentals) => ({
  type: 'GET_LOCAL_STORAGE_RENTALS',
  rentals
});

export const addRental = (item) => ({
  type: 'ADD_RENTAL',
  item
});

export const removeRental = (item) => ({
  type: 'REMOVE_RENTAL',
  item
});

export const addQuote = (deliveryQuote) => ({
  type: 'ADD_QUOTE',
  deliveryQuote
});
