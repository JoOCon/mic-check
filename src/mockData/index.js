export const mockUser = {
  userName: 'Tom Hanks',
  password: 'password',
  email: 'THanks@hanks.com',
  location: '20 McAllister St, San Francisco, CA'
};

export const mockUsers = [{
  userName: 'Tom Hanks',
  password: 'password',
  email: 'THanks@hanks.com',
  location: '20 McAllister St, San Francisco, CA'
},
{
  userName: 'Jim Hanks',
  password: 'password',
  email: 'JHanks@hanks.com',
  location: '20 McAllister St, San Francisco, CA'
}];

export const mockItem = {
  name: 'Guitar',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a guitar!',
  image: "https://guitar-image.png",
  id: 1
};

export const mockItem2 = {
  name: 'Amp',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a amp!',
  image: "https://amp-image.png",
  id: 2
};

export const mockItems = [{
  name: 'Guitar',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a guitar!',
  image: "https://guitar-image.png",
  id: 1
}, 
{
  name: 'Amp',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a amp!',
  image: "https://amp-image.png",
  id: 2
}];

export const mockSenderLocation = '101 Market St, San Francisco, CA';
export const mockDeliveryLocation = '20 McAllister St, San Francisco, CA';

export const mockQuoteData = {
  kind: "delivery_quote",
  fee: 1140,
  created: "2018-09-13T00:40:02Z",
  pickup_duration: 5,
  expires: "2018-09-13T00:45:01Z",
  currency: "usd",
  currency_type: "USD",
  duration: 45,
  dropoff_eta: "2018-09-13T01:25:02Z",
  id: "dqt_Ltalsge5ktl1YV"
};

export const mockOrderInfo = {
  manifest: 'a box of shoes',
  'manifest_items[].name': 'Nikes',
  'manifest_items[].size': '2',
  pickup_address: '101 Market St, San Francisco, CA',
  pickup_name: 'Allen',
  pickup_phone_number: 3033334444,
  dropoff_address: '20 McAllister St, San Francisco, CA',
  dropoff_name: 'Tim',
  dropoff_phone_number: 2223334444,
  dropoff_notes: 'Tim@timmmy.com'
};