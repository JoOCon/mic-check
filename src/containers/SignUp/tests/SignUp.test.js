import React from 'react';
import { shallow } from 'enzyme';
import { SignUp, mapStateToProps, mapDispatchToProps } from '../';
import * as mockData from '../../../mockData';
import * as actions from '../../../actions';

describe('SignUp tests', () => {
  let wrapper;
  let mockRunFunction;
  const { mockUser, mockUsers } = mockData;
  const { setActiveUser, addUser } = actions;
  beforeEach(() => {
    mockRunFunction = jest.fn();
    wrapper = shallow( 
      <SignUp 
        users={mockUsers}
        activeUser={mockUser}
        addUser={mockRunFunction}
        setActiveUser={mockRunFunction}
      /> 
    );
  });

  it('renders the SignUp with the correct props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have initial state', () => {
    const mockInitalState = {
      userName: '',
      email: '',
      phoneNumber: '',
      password: '',
      street: '',
      city: '',
      state: '',
      redirect: false
    };
    expect(wrapper.state()).toEqual(mockInitalState);
  });

  describe('handleChange Tests', () => {
    it('should have userName updated in state', () => {
      const mockUserNameEvent = {target: {name: 'userName', value: 'Jim'}};
      wrapper.instance().handleChange(mockUserNameEvent);
      const expected = mockUserNameEvent.target.value;
      const result = wrapper.state('userName');
      expect(result).toEqual(expected);
    });

    it('should have Email updated in state', () => {
      const mockEmailEvent = {target: {name: 'email', value: 'j@j'}};
      wrapper.instance().handleChange(mockEmailEvent);
      const expected = mockEmailEvent.target.value;
      const result = wrapper.state('email');
      expect(result).toEqual(expected);
    });

    it('should have phoneNumber updated in state', () => {
      const mockPhoneNumberEvent = {target: {name: 'phoneNumber', value: 3033334444}};
      wrapper.instance().handleChange(mockPhoneNumberEvent);
      const expected = mockPhoneNumberEvent.target.value;
      const result = wrapper.state('phoneNumber');
      expect(result).toEqual(expected);
    });

    it('should have password updated in state', () => {
      const mockPasswordEvent = {target: {name: 'password', value: 'j'}};
      wrapper.instance().handleChange(mockPasswordEvent);
      const expected = mockPasswordEvent.target.value;
      const result = wrapper.state('password');
      expect(result).toEqual(expected);
    });

    it('should have street updated in state', () => {
      const mockStreetEvent = {target: {name: 'street', value: '234 ivy st.'}};
      wrapper.instance().handleChange(mockStreetEvent);
      const expected = mockStreetEvent.target.value;
      const result = wrapper.state('street');
      expect(result).toEqual(expected);
    });

    it('should have city updated in state', () => {
      const mockCityEvent = {target: {name: 'city', value: 'Denver'}};
      wrapper.instance().handleChange(mockCityEvent);
      const expected = mockCityEvent.target.value;
      const result = wrapper.state('city');
      expect(result).toEqual(expected);
    });

    it('should have state updated in state', () => {
      const mockStateEvent = {target: {name: 'state', value: 'CO'}};
      wrapper.instance().handleChange(mockStateEvent);
      const expected = mockStateEvent.target.value;
      const result = wrapper.state('state');
      expect(result).toEqual(expected);
    });
  });

  describe('MapStateToProps', () => {
    it('should have a users data object in props', () => {
      const mockState = {users: mockUser};
      const expected = {users: mockUser};
      const props = mapStateToProps(mockState);
      expect(props).toEqual(expected);
    });

    it('should have a activeUser data object in props', () => {
      const mockState = {activeUser: mockUser};
      const expected = {activeUser: mockUser};
      const props = mapStateToProps(mockState);
      expect(props).toEqual(expected);
    });
  });

  describe('MapDispatchToProps', () => {
    let mockDispatch;
    beforeEach(() => {
      mockDispatch = jest.fn();
    });

    it('should dispatch addUser when called', () => {
      const mockAction = addUser(mockUser);
      const props = mapDispatchToProps(mockDispatch);
      props.addUser(mockUser);
      expect(mockDispatch).toBeCalledWith(mockAction);
    });

    it('should dispatch setActiveUser when called', () => {
      const mockAction = setActiveUser(mockUser);
      const props = mapDispatchToProps(mockDispatch);
      props.setActiveUser(mockUser);
      expect(mockDispatch).toBeCalledWith(mockAction);
    });
  });
});
