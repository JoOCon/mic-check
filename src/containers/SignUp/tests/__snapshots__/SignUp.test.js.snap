// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp tests renders the SignUp with the correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    activeUser={
        Object {
            "email": "THanks@hanks.com",
            "location": "20 McAllister St, San Francisco, CA",
            "password": "password",
            "userName": "Tom Hanks",
          }
    }
    addUser={[Function]}
    setActiveUser={[Function]}
    users={
        Array [
            Object {
              "email": "THanks@hanks.com",
              "location": "20 McAllister St, San Francisco, CA",
              "password": "password",
              "userName": "Tom Hanks",
            },
            Object {
              "email": "JHanks@hanks.com",
              "location": "20 McAllister St, San Francisco, CA",
              "password": "password",
              "userName": "Jim Hanks",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          className="user-signup"
          onSubmit={[Function]}
>
          <h2
                    className="log-header"
          >
                    Sign Up
          </h2>
          <input
                    className="input-field"
                    name="userName"
                    onChange={[Function]}
                    placeholder="username"
                    value=""
          />
          <input
                    className="input-field"
                    name="email"
                    onChange={[Function]}
                    placeholder="email"
                    type="email"
                    value=""
          />
          <input
                    className="input-field"
                    name="phoneNumber"
                    onChange={[Function]}
                    placeholder="Phone Number"
                    type="phoneNumber"
                    value=""
          />
          <input
                    className="input-field"
                    name="password"
                    onChange={[Function]}
                    placeholder="password"
                    type="password"
                    value=""
          />
          <input
                    className="input-field"
                    name="street"
                    onChange={[Function]}
                    placeholder="street"
                    value=""
          />
          <input
                    className="input-field"
                    name="city"
                    onChange={[Function]}
                    placeholder="city"
                    value=""
          />
          <input
                    className="input-field"
                    name="state"
                    onChange={[Function]}
                    placeholder="state"
                    value=""
          />
          <button
                    className="signup-button"
          >
                    SignUp
          </button>
</form>,
        false,
        <Link
          replace={false}
          to="/Login"
>
          <button
                    className="login"
          >
                    Login
          </button>
</Link>,
        <Link
          replace={false}
          to="/"
>
          <button
                    className="home"
          >
                    Home
          </button>
</Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="log-header"
>
              Sign Up
</h2>,
            <input
              className="input-field"
              name="userName"
              onChange={[Function]}
              placeholder="username"
              value=""
/>,
            <input
              className="input-field"
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="email"
              value=""
/>,
            <input
              className="input-field"
              name="phoneNumber"
              onChange={[Function]}
              placeholder="Phone Number"
              type="phoneNumber"
              value=""
/>,
            <input
              className="input-field"
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
/>,
            <input
              className="input-field"
              name="street"
              onChange={[Function]}
              placeholder="street"
              value=""
/>,
            <input
              className="input-field"
              name="city"
              onChange={[Function]}
              placeholder="city"
              value=""
/>,
            <input
              className="input-field"
              name="state"
              onChange={[Function]}
              placeholder="state"
              value=""
/>,
            <button
              className="signup-button"
>
              SignUp
</button>,
          ],
          "className": "user-signup",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "className": "log-header",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "userName",
              "onChange": [Function],
              "placeholder": "username",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "email",
              "onChange": [Function],
              "placeholder": "email",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "phoneNumber",
              "onChange": [Function],
              "placeholder": "Phone Number",
              "type": "phoneNumber",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "password",
              "onChange": [Function],
              "placeholder": "password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "street",
              "onChange": [Function],
              "placeholder": "street",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "city",
              "onChange": [Function],
              "placeholder": "city",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field",
              "name": "state",
              "onChange": [Function],
              "placeholder": "state",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SignUp",
              "className": "signup-button",
            },
            "ref": null,
            "rendered": "SignUp",
            "type": "button",
          },
        ],
        "type": "form",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="login"
>
            Login
</button>,
          "replace": false,
          "to": "/Login",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "className": "login",
          },
          "ref": null,
          "rendered": "Login",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="home"
>
            Home
</button>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Home",
            "className": "home",
          },
          "ref": null,
          "rendered": "Home",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="user-signup"
            onSubmit={[Function]}
>
            <h2
                        className="log-header"
            >
                        Sign Up
            </h2>
            <input
                        className="input-field"
                        name="userName"
                        onChange={[Function]}
                        placeholder="username"
                        value=""
            />
            <input
                        className="input-field"
                        name="email"
                        onChange={[Function]}
                        placeholder="email"
                        type="email"
                        value=""
            />
            <input
                        className="input-field"
                        name="phoneNumber"
                        onChange={[Function]}
                        placeholder="Phone Number"
                        type="phoneNumber"
                        value=""
            />
            <input
                        className="input-field"
                        name="password"
                        onChange={[Function]}
                        placeholder="password"
                        type="password"
                        value=""
            />
            <input
                        className="input-field"
                        name="street"
                        onChange={[Function]}
                        placeholder="street"
                        value=""
            />
            <input
                        className="input-field"
                        name="city"
                        onChange={[Function]}
                        placeholder="city"
                        value=""
            />
            <input
                        className="input-field"
                        name="state"
                        onChange={[Function]}
                        placeholder="state"
                        value=""
            />
            <button
                        className="signup-button"
            >
                        SignUp
            </button>
</form>,
          false,
          <Link
            replace={false}
            to="/Login"
>
            <button
                        className="login"
            >
                        Login
            </button>
</Link>,
          <Link
            replace={false}
            to="/"
>
            <button
                        className="home"
            >
                        Home
            </button>
</Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="log-header"
>
                Sign Up
</h2>,
              <input
                className="input-field"
                name="userName"
                onChange={[Function]}
                placeholder="username"
                value=""
/>,
              <input
                className="input-field"
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="email"
                value=""
/>,
              <input
                className="input-field"
                name="phoneNumber"
                onChange={[Function]}
                placeholder="Phone Number"
                type="phoneNumber"
                value=""
/>,
              <input
                className="input-field"
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="password"
                value=""
/>,
              <input
                className="input-field"
                name="street"
                onChange={[Function]}
                placeholder="street"
                value=""
/>,
              <input
                className="input-field"
                name="city"
                onChange={[Function]}
                placeholder="city"
                value=""
/>,
              <input
                className="input-field"
                name="state"
                onChange={[Function]}
                placeholder="state"
                value=""
/>,
              <button
                className="signup-button"
>
                SignUp
</button>,
            ],
            "className": "user-signup",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
                "className": "log-header",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "name": "userName",
                "onChange": [Function],
                "placeholder": "username",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "name": "email",
                "onChange": [Function],
                "placeholder": "email",
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "name": "phoneNumber",
                "onChange": [Function],
                "placeholder": "Phone Number",
                "type": "phoneNumber",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "name": "password",
                "onChange": [Function],
                "placeholder": "password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "name": "street",
                "onChange": [Function],
                "placeholder": "street",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "name": "city",
                "onChange": [Function],
                "placeholder": "city",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input-field",
                "name": "state",
                "onChange": [Function],
                "placeholder": "state",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "SignUp",
                "className": "signup-button",
              },
              "ref": null,
              "rendered": "SignUp",
              "type": "button",
            },
          ],
          "type": "form",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="login"
>
              Login
</button>,
            "replace": false,
            "to": "/Login",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
              "className": "login",
            },
            "ref": null,
            "rendered": "Login",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="home"
>
              Home
</button>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Home",
              "className": "home",
            },
            "ref": null,
            "rendered": "Home",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
